actors/cloudbroker/cloudbroker__computenode/methodclass/cloudbroker_computenode.py

```diff
@@ -124,11 +123,20 @@ def enable(self, id, message, **kwargs):
 
     @auth(['level2', 'level3'], True)
     def upgrade(self, id, message, force, **kwargs):
+        kwargs['ctx'].events.runAsync(self._upgradeStack,
+                                      args=(id, message, force),
+                                      kwargs=kwargs,
+                                      title='upgrading Stack',
+                                      success='Successfully Upgraded stack',
+                                      error='Failed to Upgrade Stacks')
```

Comment:
it might be usefull to add a message here telling the user the machine is being upgraded. Also make sure in the portal so who the response (there is a flag for this in the actions macro)